backend/src/routes/api/connection-types/connectionTypeUtils.ts

```diff
@@ -0,0 +1,191 @@
+
+  if (labels[ODH_DASHBOARD_KEY] !== 'true' || labels[CONNECTION_TYPE_KEY] !== 'true') {
+    const error = 'Unable to add connection type, incorrect labels.';
+    fastify.log.error(error);
+    return { success: false, error };
+  }
+
+  try {
+    const validConnectionType = await isExistingConnectionType(fastify, name);
+    if (!validConnectionType) {
+      const error = `Unable to update connection type, object is not a connection type`;
+      fastify.log.error(error);
+      return { success: false, error };
+    }
+
+    await coreV1Api.replaceNamespacedConfigMap(name, dashboardNamespace, connectionType);
+    return { success: true, error: '' };
+  } catch (e) {
+    const error = `Unable to update connection type: ${e.message}.`;
+    fastify.log.error(error);
+    return { success: false, error };
+  }
+};
+
+export const patchConnectionType = async (
+  fastify: KubeFastifyInstance,
+  name: string,
+  connectionType: V1ConfigMap,
+): Promise<{ success: boolean; error: string }> => {
+  const coreV1Api = fastify.kube.coreV1Api;
+  const labels = connectionType.metadata.labels;
+  const { dashboardNamespace } = getNamespaces(fastify);
+
+  if (labels && (labels[ODH_DASHBOARD_KEY] !== 'true' || labels[CONNECTION_TYPE_KEY] !== 'true')) {
+    const error = 'Unable to update connection type, incorrect labels.';
+    fastify.log.error(error);
+    return { success: false, error };
+  }
+
+  try {
+    const validConnectionType = await isExistingConnectionType(fastify, name);
+    if (!validConnectionType) {
+      const error = `Unable to update connection type, object is not a connection type`;
+      fastify.log.error(error);
+      return { success: false, error };
+    }
+
+    await coreV1Api.patchNamespacedConfigMap(
+      name,
+      dashboardNamespace,
+      connectionType,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      {
+        headers: { 'Content-Type': 'application/merge-patch+json' },
+      },
+    );
+    return { success: true, error: '' };
+  } catch (e) {
+    const error = `Unable to update connection type: ${e.message}.`;
+    fastify.log.error(error);
+    return { success: false, error };
+  }
+};
+
+export const deleteConnectionType = async (
+  fastify: KubeFastifyInstance,
+  name: string,
+): Promise<{ success: boolean; error: string }> => {
+  const { dashboardNamespace } = getNamespaces(fastify);
+  const coreV1Api = fastify.kube.coreV1Api;
+  try {
+    const validConnectionType = await isExistingConnectionType(fastify, name);
+    if (!validConnectionType) {
+      const error = `Unable to delete connection type, object is not a connection type`;
+      fastify.log.error(error);
+      return { success: false, error };
+    }
+    await coreV1Api.deleteNamespacedConfigMap(name, dashboardNamespace);
+    return { success: true, error: '' };
+  } catch (e) {
+    const error = `Unable to delete connection type: ${e.message}.`;
```

Comment:
Use `errorHandler` as we will eventually have stricter ts / eslint rules.
```suggestion
    const error = `Unable to delete connection type: ${errorHandler(e)}.`;
```